File: WebAPI/Models/DbModels/ListType.cs
using System.ComponentModel.DataAnnotations.Schema;
using WebAPI.Models.DbModels.Interfaces;

namespace WebAPI.Models.DbModels
{
    public class ListType : IDbItem
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.Now;
        public DateTime LastUpdatedDate { get; set; } = DateTime.Now;
        public string Name { get; set; }
        public string Description { get; set; } = "";
        public List<ListItem> ListItems { get; set; }
        public ListType() { }
        public ListType(string Name, string Description = "") 
        {
            this.Name = Name;
        }

    }
}

File: WebApp/src/app/models/listType.ts
import { ListItem } from './listItem';

export class ListType {
  id: string;
  createdDate: Date;
  lastUpdatedDate: Date;
  name: string;
  description: string;
  listItems: ListItem[];
  constructor(obj?: ListType) {
    this.id = obj?.id || '';
    this.createdDate = obj?.createdDate || new Date();
    this.lastUpdatedDate = obj?.lastUpdatedDate || new Date();
    this.name = obj?.name || 'Default';
    this.description = obj?.description || '';
    this.listItems = obj?.listItems || [];
  }
}
